src/cplscheme/BaseCouplingScheme.hpp

```diff
@@ -225,20 +213,28 @@ class BaseCouplingScheme : public CouplingScheme {
   void setIterationAcceleration(acceleration::PtrAcceleration acceleration);
 
 protected:
-  /// Sets whether explicit or implicit coupling is being done.
-  CouplingMode _couplingMode = Undefined;
+  /// Returns true, if coupling scheme is explicit
+  bool isExplicitCouplingScheme()
+  {
+    PRECICE_CHECK(_couplingMode != Undefined, "Undefined coupling mode is not allowed.");
```

Comment:
```suggestion
    PRECICE_ASSERT(_couplingMode != Undefined);
```

Only use checks for states a user can reach.